src/dbwrapper.h

```diff
@@ -16,23 +16,39 @@
 #include <leveldb/db.h>
 #include <leveldb/write_batch.h>
 
-class leveldb_error : public std::runtime_error
+class dbwrapper_error : public std::runtime_error
 {
 public:
-    leveldb_error(const std::string& msg) : std::runtime_error(msg) {}
+    dbwrapper_error(const std::string& msg) : std::runtime_error(msg) {}
 };
 
+class CDBWrapper;
+
+/** These should be considered an implementation detail of the specific database.
+ */
+namespace dbwrapper_private {
+
+/** Handle database error by throwing dbwrapper_error exception.
+ */
 void HandleError(const leveldb::Status& status);
 
-/** Batch of changes queued to be written to a CLevelDBWrapper */
-class CLevelDBBatch
+};
+
+/** Batch of changes queued to be written to a CDBWrapper */
+class CDBBatch
 {
-    friend class CLevelDBWrapper;
+    friend class CDBWrapper;
 
 private:
+    const CDBWrapper &parent;
```

Comment:
Is it me or this field is never used?

Reply:
Yes, good catch. There is one also in `CDBIterator`.
Those were only used to get the key for the data obfuscation. We can remove both of them if we don't plan to introduce this feat.